magentoerpconnect_rma/claim.py

```diff
@@ -556,45 +575,113 @@ def crm_claim_import_batch(session, model_name, backend_id, filters=None):
         magentoerpconnect.delay_export_all_bindings(
             session, model_name, record_id, vals=vals)
 
 
-@magento
+@magento2000
 class MagentoCrmClaimExporter(MagentoExporter):
-    """ Export claim state to Magento. """
     _model_name = ['magento.crm.claim']
 
     def _should_import(self):
         return False
 
 
-@magento
+@magento2000
 class MagentoCrmClaimExportMapper(ExportMapper):
     _model_name = 'magento.crm.claim'
 
     @mapping
-    def stage_id(self, record):
-        state = record.stage_id.name
-        return {'state': state}
+    def items(self, record):
+        """Export the quantity actually returned by the customer."""
+        cr, uid = self.session.cr, self.session.uid
+        imd_model = self.session.pool["ir.model.data"]
+        stage_approved = imd_model.get_object(
+            cr, uid, "crm_claim", "stage_claim2")
+        order_binder = self.get_binder_for_model("magento.sale.order")
+        order_line_binder = self.get_binder_for_model("magento.sale.order.line")
+        magento_order_id = order_binder.to_backend(
+            record.order_id.id, wrap=True)
+        res = {
+            "entity_id": record.magento_id,
+            # 'increment_id', 'order_id' required by Magento
+            "increment_id": record.number,
+            "order_id": magento_order_id,
+            "items": [],
+        }
+        for line in record.magento_claim_line_ids:
+            magento_order_line_id = order_line_binder.to_backend(
+                line.order_line_id.id, wrap=True)
+            item = {
+                "entity_id": line.magento_id,
+                "rma_entity_id": record.magento_id,
+                # 'order_item_id' required by Magento
+                "order_item_id": magento_order_line_id,
+                "qty_returned": line.product_returned_quantity,
+                "reason": line.reason,
+                "resolution": line.resolution,
+            }
+            # Set the approved quantity when the claim is validated/approved
+            if record.stage_id == stage_approved:
+                item["qty_approved"] = item["qty_returned"]
+            res["items"].append(item)
+        res = self._set_status(record, res)
+        if not res:
+            raise orm.except_orm(
+                u"Mapping Error",
+                u"Unable to map the stage '%s' for Magento.\n"
+                u"You can overload the '_set_status' method to extend "
+                u"the mapping.")
+        return res
+
+    def _set_status(self, record, res):
+        """Update `res` to set a valid Magento status."""
+        cr, uid = self.session.cr, self.session.uid
+        stage_model = self.session.pool["crm.claim.stage"]
+        stage_id = record.stage_id.id
+        external_id = stage_model.get_external_id(cr, uid, [stage_id])[stage_id]
+        status_mapping = self._get_magento_status_mapping()
+        if not external_id or external_id not in status_mapping:
+            return False
```

Comment:
No raise in this case?

Reply:
No I choose to put the raise here: https://github.com/camptocamp/connector-magento/pull/4/files#diff-672b8f0849537bcd9cbe2cc381266fa8R631
This way the user could overload this `_set_status` method to handle its specific needs for custom stages without raising exception when calling `super()`.